apps/zbugs/shared/mutators.ts

```diff
@@ -0,0 +1,218 @@
+import {schema} from './schema.ts';
+import {must} from '../../../packages/shared/src/must.ts';
+import {assert} from '../../../packages/shared/src/asserts.ts';
+import type {UpdateValue, Transaction, CustomMutatorDefs} from '@rocicorp/zero';
+import {
+  assertIsCreatorOrAdmin as assertIsAdminOrCreator,
+  assertUserCanSeeComment,
+  assertUserCanSeeIssue,
+  isAdmin,
+  verifyToken,
+} from './validators.ts';
+
+type AddEmojiArgs = {
+  id: string;
+  unicode: string;
+  annotation: string;
+  subjectID: string;
+  creatorID: string;
+  created: number;
+};
+
+export const mutators = {
+  issue: {
+    async create(
+      tx,
+      {
+        id,
+        title,
+        description,
+        created,
+        modified,
+      }: {
+        id: string;
+        title: string;
+        description?: string;
+        created: number;
+        modified: number;
+      },
+    ) {
+      if (tx.location === 'server') {
+        created = modified = Date.now();
+      }
+      const creatorID = must((await verifyToken(tx)).sub);
+
+      await tx.mutate.issue.insert({
+        id,
+        title,
+        description: description ?? '',
+        created,
+        creatorID,
+        modified,
+        open: true,
+        visibility: 'public',
+      });
+    },
+
+    async update(tx, change: UpdateValue<typeof schema.tables.issue>) {
+      await assertIsAdminOrCreator(tx, tx.query.issue, change.id);
```

Comment:
how does the app get a new auth token.

Reply:
today `mutagen` throws an auth error that the zero-client handles, allowing the developer to refresh the auth token.

We'll need to add an error type for the custom mutator protocol to indicate auth problems.